### ISSUE TITLE
features.h not found (Macosx build)

### ISSUE THREAD

**Issue by rete** [travis-ci/travis-ci#8148]
Hello Travis devs,

I'm using Travis to build my software on Macosx. I have a build fail that shows an error from XCode :

```shell
[ 66%] Building CXX object source/dim/CMakeFiles/DimBridge.dir/src/util/dimbridge.cxx.o

cd /Users/travis/build/DQM4HEP/dqm4hep-net/build/source/dim && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++    -I/Users/travis/build/DQM4HEP/dqm4hep-net/source/dim/include -I/usr/local/include  -std=c++11  -DMIPSEL -DPROTOCOL=1 -Dunix -Dlinux -Ddarwin -fPIC -O2 -g -DNDEBUG   -o CMakeFiles/DimBridge.dir/src/util/dimbridge.cxx.o -c /Users/travis/build/DQM4HEP/dqm4hep-net/source/dim/src/util/dimbridge.cxx

In file included from /Users/travis/build/DQM4HEP/dqm4hep-net/source/dim/src/util/dimbridge.cxx:4:

In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iostream:37:

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__config:356:10: fatal error: 'features.h' file not found

#include <features.h>

         ^

1 error generated.

make[2]: *** [source/dim/CMakeFiles/DimBridge.dir/src/util/dimbridge.cxx.o] Error 1

make[1]: *** [source/dim/CMakeFiles/DimBridge.dir/all] Error 2

make: *** [all] Error 2
```

Here is the the build job :
https://travis-ci.org/DQM4HEP/dqm4hep-net/jobs/256698283

I found no other issues I can refer to, either from google or the travis issue page.

**Comment by BanzaiMan** [comment]
I do not think `features.h` exists for OS X. You'd have to come up with a way to emulate it on the Mac.

**Comment by rete** [comment]
Yes the fact is that I'm not including it. I'm including iostream that includes features.h

```shell
In file included from /Users/travis/build/DQM4HEP/dqm4hep-net/source/dim/src/util/dimbridge.cxx:4:

In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iostream:37:

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__config:356:10: fatal error: 'features.h' file not found

#include <features.h>
```

**Comment by BanzaiMan** [comment]
Oh, I see. So, this is an issue with Apple's packaging, right? It is `include`ing the file it does not have.

**Comment by rete** [comment]
Yes, it seems that the packaging does not provide this header which is included in iostream, which may be a frequent problem. I'm actually suprised that there's no other reference to this issue somewhere in the bug tracker as the problem is coming from iostream.

**Comment by melroy89** [comment]
Hi. I have a similar issue on another project.

What was the final solution here??

https://github.com/LibreWeb/Browser/runs/5079105213?check_suite_focus=true

```
/Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DJSON_DIAGNOSTICS=0 -DJSON_USE_IMPLICIT_CONVERSIONS=1 -D_DARWIN_C_SOURCE -D_XOPEN_SOURCE=500 -D__linux__ -I/Users/runner/work/Browser/Browser/build/src -I/Users/runner/work/Browser/Browser/src -I/Users/runner/work/Browser/Browser/build/lib/commonmarker/src -I/Users/runner/work/Browser/Browser/build/lib/commonmarker/extensions -I/usr/local/Cellar/libffi/3.4.2/include -I/usr/local/Cellar/gtkmm3/3.24.5/include/gtkmm-3.0 -I/usr/local/Cellar/gtkmm3/3.24.5/lib/gtkmm-3.0/include -I/usr/local/Cellar/atkmm@2.28/2.28.2/include/atkmm-1.6 -I/usr/local/Cellar/atkmm@2.28/2.28.2/lib/atkmm-1.6/include -I/usr/local/Cellar/gtk+3/3.24.30_1/include/gtk-3.0/unix-print -I/usr/local/Cellar/gtkmm3/3.24.5/include/gdkmm-3.0 -I/usr/local/Cellar/gtkmm3/3.24.5/lib/gdkmm-3.0/include -I/usr/local/Cellar/glibmm@2.66/2.66.2/include/giomm-2.4 -I/usr/local/Cellar/glibmm@2.66/2.66.2/lib/giomm-2.4/include -I/usr/local/Cellar/gtk+3/3.24.30_1/include/gtk-3.0 -I/usr/local/Cellar/glib/2.70.3/include/gio-unix-2.0 -I/usr/local/Cellar/cairo/1.16.0_5/include/cairo -I/usr/local/Cellar/libepoxy/1.5.9/include -I/usr/local/Cellar/atk/2.36.0/include/atk-1.0 -I/usr/local/Cellar/pangomm@2.46/2.46.2/include/pangomm-1.4 -I/usr/local/Cellar/pangomm@2.46/2.46.2/lib/pangomm-1.4/include -I/usr/local/Cellar/glibmm@2.66/2.66.2/include/glibmm-2.4 -I/usr/local/Cellar/glibmm@2.66/2.66.2/lib/glibmm-2.4/include -I/usr/local/Cellar/cairomm@1.14/1.14.3/include/cairomm-1.0 -I/usr/local/Cellar/cairomm@1.14/1.14.3/lib/cairomm-1.0/include -I/usr/local/Cellar/libsigc++@2/2.10.7/include/sigc++-2.0 -I/usr/local/Cellar/libsigc++@2/2.10.7/lib/sigc++-2.0/include -I/usr/local/Cellar/pango/1.50.3/include/pango-1.0 -I/usr/local/Cellar/harfbuzz/3.3.1/include/harfbuzz -I/usr/local/Cellar/fribidi/1.0.11/include/fribidi -I/usr/local/Cellar/graphite2/1.3.14/include -I/usr/local/Cellar/pixman/0.40.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.37/include/libpng16 -I/usr/local/Cellar/libxcb/1.14_1/include -I/usr/local/Cellar/libxrender/0.9.10/include -I/usr/local/Cellar/libxext/1.3.4/include -I/usr/local/Cellar/libx11/1.7.3.1/include -I/usr/local/Cellar/libxau/1.0.9/include -I/usr/local/Cellar/libxdmcp/1.1.3/include -I/usr/local/Cellar/gdk-pixbuf/2.42.6/include/gdk-pixbuf-2.0 -I/usr/local/Cellar/libtiff/4.3.0/include -I/usr/local/Cellar/glib/2.70.3/include -I/usr/local/Cellar/glib/2.70.3/include/glib-2.0 -I/usr/local/Cellar/glib/2.70.3/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.45/include -I/usr/local/Cellar/xorgproto/2021.5/include -I/Users/runner/work/Browser/Browser/lib/commonmarker/src -I/Users/runner/work/Browser/Browser/lib/commonmarker/extensions -I/Users/runner/work/Browser/Browser/lib/ipfs-http-client/include -I/Users/runner/work/Browser/Browser/build/_deps/json-src/single_include -isystem /Users/runner/work/Browser/Browser/lib/whereami -Wall -Wextra -Werror -isysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.6 -std=c++2a -MD -MT src/CMakeFiles/libreweb-browser.dir/main.cc.o -MF src/CMakeFiles/libreweb-browser.dir/main.cc.o.d -o src/CMakeFiles/libreweb-browser.dir/main.cc.o -c /Users/runner/work/Browser/Browser/src/main.cc
In file included from /Users/runner/work/Browser/Browser/src/main.cc:1:
In file included from /Users/runner/work/Browser/Browser/src/ipfs-daemon.h:4:
In file included from /usr/local/Cellar/glibmm@2.66/2.66.2/include/glibmm-2.4/glibmm/spawn.h:24:
In file included from /usr/local/Cellar/glibmm@2.66/2.66.2/include/glibmm-2.4/glibmm/arrayhandle.h:21:
In file included from /usr/local/Cellar/glibmm@2.66/2.66.2/include/glibmm-2.4/glibmm/containerhandle_shared.h:21:
In file included from /usr/local/Cellar/glibmm@2.66/2.66.2/include/glibmm-2.4/glibmm/refptr.h:21:
In file included from /usr/local/Cellar/glib/2.70.3/include/glib-2.0/glib.h:30:
In file included from /usr/local/Cellar/glib/2.70.3/include/glib-2.0/glib/galloca.h:32:
In file included from /usr/local/Cellar/glib/2.70.3/include/glib-2.0/glib/gtypes.h:32:
In file included from /usr/local/Cellar/glib/2.70.3/lib/glib-2.0/include/glibconfig.h:9:
In file included from /usr/local/Cellar/glib/2.70.3/include/glib-2.0/glib/gmacros.h:38:
In file included from /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/stddef.h:39:
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/__config:279:12: fatal error: 'features.h' file not found
#  include <features.h>
           ^~~~~~~~~~~~
```
